### DISCUSSION TITLE
How to yank then paste line by line

### DISCUSSION THREAD

**Discussion by dragove** [meow-edit/meow#439] [Q&A] [answered]
I'm exploring meow from vim.
in vim you can use C-v block visual mode to copy a block of text and then paste them in another block. For example in text
```
one
three
five

two
four
six
```
I can copy the first three lines as block and then paste in front of last three lines. and then text will be merged to
```
one   two
three four
five  six
```

I wonder if I can achieve the same thing using meow.

**Comment by Linerre** [comment, marked as answer]
In Emacs terms, what you want to achieve is called "Rectangles" (see [12.5 Rectangles](https://www.gnu.org/software/emacs/manual/html_node/emacs/Rectangles.html)).

I doubt that Meow alone can do this (as it is not particularly designed for such tasks). But Meow does _not_ modify Emacs's default key bindings, so you can rely on Emacs's default way:
1. Move your point (cursor) to the beginning of word `one`
2. Press `C-x SPC` to activate `rectangle-mark-mode`
3. Keep pressing `C-n` (3 times) to move the point to letter `f` of `five`
4. Keep pressing `C-f` (or `C-e` then `C-f`) to expand the *rectangular* region to include all the three words and *one* space character after `three`:
``` 
one   |
three |
five  |
```
In the above text area, `|` represents the rectangle's right margin. In Emacs, this selected/marked region should have been highlighted.
5. Press `M-w` to copy the selected region.
6. Move to `t` of word `two`
7. Press `p` (or whatever you have bound `meow-yank` to)
8. Done

**Reply by dragove** [reply]
Oh, this works for me, thanks, And I found meow motion won't break rectangle-mark-mode, meow-save and meow-yank is also working, I can just bind key to this mode and use it like vim. Many thanks for your detailed solution.
